fgpyo/sam/builder.py

```diff
@@ -503,34 +503,49 @@ def to_path(
         path: Optional[Path] = None,
         index: bool = True,
         pred: Callable[[AlignedSegment], bool] = lambda r: True,
+        tmp_file_type: Optional[sam.SamFileType] = None,
     ) -> Path:
         """Write the accumulated records to a file, sorts & indexes it, and returns the Path.
         If a path is provided, it will be written to, otherwise a temporary file is created
         and returned.
 
+        If `path` is provided, `tmp_file_type` may not be provided. In this case, the file type
+        (SAM/BAM/CRAM) will be automatically determined by the file extension when a path
+        is provided.  See `~pysam` for more details.
+
+        If `path` is not provided, the file type will default to BAM unless `tmp_file_type` is
+        provided.
+
         Args:
             path: a path at which to write the file, otherwise a temp file is used.
-            index: if True and `sort_order` is `Coordinate` index is generated, otherwise not.
+            index: if True and `sort_order` is `Coordinate` and output is a BAM file, then
+                   an index is generated, otherwise not.
```

Comment:
```suggestion
            index: if True and `sort_order` is `Coordinate` and output is a BAM/CRAM file,
                   then an index is generated, otherwise not.
```